hq/app/schedule/IamNotifier.scala

```diff
@@ -3,36 +3,54 @@ package schedule
 import com.amazonaws.services.sns.AmazonSNSAsync
 import com.gu.anghammarad.Anghammarad
 import com.gu.anghammarad.models.{Email, Notification, Preferred, Target}
-import model.AwsAccount
+import model._
+import org.joda.time.DateTime
 import play.api.Logging
 import schedule.IamMessages.{sourceSystem, subject}
+import utils.attempt.{Attempt, FailedAttempt}
 
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 
 object IamNotifier extends Logging {
   val channel = Preferred(Email)
 
-  def createNotification(accountName: AwsAccount, targets: List[Target], message: String): Notification = {
-    Notification(subject(accountName), message, List.empty, targets, channel, sourceSystem)
+  def createNotification(
+    accountName: AwsAccount,
+    targets: List[Target],
+    message: String,
+    awsAccountId: String,
+    username: String,
+    emailType: IamAuditNotificationType = Warning
```

Comment:
The functionality that determines whether a notification is a `Warning` or a `Final` notification will be created later, so this is set as `Warning` for now.